hazelcast/src/main/java/com/hazelcast/map/impl/recordstore/DefaultRecordStore.java

```diff
@@ -863,59 +862,60 @@ public Object put(Data key, Object value, long ttl, long maxIdle) {
     @SuppressWarnings({"checkstyle:npathcomplexity",
             "checkstyle:parameternumber", "checkstyle:cyclomaticcomplexity"})
     private Object putInternal(Data key, Object newValue, long ttl,
-                               long maxIdle, long expiryTime, long now, boolean load, boolean store,
-                               boolean putIfAbsent, boolean putIfExists, boolean putIfEqual,
-                               boolean putFromLoad, Object expectedValue, boolean setTtl,
-                               boolean checkIfLoaded, @Nullable UUID transactionId,
-                               Address callerAddress, boolean countAsAccess, boolean backup) {
+                               long maxIdle, long expiryTime, long now, Object expectedValue,
+                               @Nullable UUID transactionId, Address callerAddress,
+                               StaticParams staticParams) {
         // If this method has to wait end of map loading.
-        if (checkIfLoaded) {
+        if (staticParams.isCheckIfLoaded()) {
             checkIfLoaded();
         }
 
         Object oldValue = null;
 
         // Get record by checking expiry, if expired, evict record.
-        Record record = getRecordOrNull(key, now, backup);
+        Record record = getRecordOrNull(key, now, staticParams.isBackup());
 
         // Variants of loading oldValue
-        if (!putIfAbsent && !putIfExists) {
+        if (staticParams.isPutVanilla()) {
```

Comment:
If I understand `vanilla` right here and it is used to mark the `!putIfAbsent && !putIfExists` cases, I'm not sure if it makes easier or more difficult to read the code, because this information will be lost after the merge. The reader will need to deduce this fact from the `else if`s, which use these two checks explicitly. I would remove `is/setPutVanilla` to make this more straightforward to read.

Reply:
Makes sense but given that this PR is the start of the PRs for HZ-1007, i will be also separating this if-else flow in subsequent PRs and in that case the benefit will be much clear i hope.

Reply:
Makes sense :+1: